naloge/2016/dn1/matrix/JureMarkun/cheapmatrix.py

```diff
@@ -0,0 +1,180 @@
+            A12 = left[0:n2//2,(m2//2):m2]
+            A21 = left[n2//2:n2,0:m2//2]
+            A22 = left[n2//2:n2,m2//2:m2]
+
+            B11 = right[0:(m2//2),0:(l2//2)] #razdelitev bločnega dela desnega matrike na štiri enako veliko matrike
+            B12 = right[0:m2//2,(l2//2):l2]
+            B21 = right[m2//2:m2,0:l2//2]
+            B22 = right[m2//2:m2,l2//2:l2]
+
+            #Na tej točki se začne koda cheapmatrix razlikovati od fastmatrix
+
+            W11 = work[0:(n2 // 2), 0:(l2 // 2)]
+            W12 = work[0:(n2 // 2), (l2 // 2):l2]
+            W21 = work[(n2 // 2):n2, 0:(l2 // 2)]
+            W22 = work[(n2 // 2):n2, (l2 // 2):l2]
+
+            work[0:work.nrow(), 0:work.ncol()] = 0
+            #Sedaj bomo vsako Matriko Mi posebej izračunali, saj ne smemo uporabiti seštevanja dveh matrik, tako kot prej, ker to ustvari novo matriko.
+
+            #M7
+            CheapMatrix.multiply(W11,(A12 - A22),(B21 + B22),W12)
+            W12[0:(n2 // 2), 0:(l2 // 2)] = 0
+            self += work
+            work[0:n2, 0:l2] = 0
+
+            #M6
+            CheapMatrix.multiply(W22,(A21 - A11), (B11 + B12),W21)
+            W21[:(n2 // 2), 0:(l2 // 2)] = 0
+            self += work
+            work[0:n2, 0:l2] = 0
+
+
+            #M4
+            CheapMatrix.multiply(W11,(A22), (B21 - B11), W21)
+            W21[0:(n2 // 2), 0:(l2 // 2)] = W11[0:(n2 // 2), 0:(l2 // 2)]
+            self += work
+            work[0:n2, 0:l2] = 0
+
+            #M3
+            CheapMatrix.multiply(W12,(A11), (B12 - B22), W22)
+            W22[0:(n2 // 2), 0:(l2 // 2)] = W12[0:(n2 // 2), 0:(l2 // 2)]
+            self += work
+            work[0:n2, 0:l2] = 0
+
+            #M5
+            CheapMatrix.multiply(W12,(A11 + A12), (B22), W11)
+            W11[0:(n2 // 2), 0:(l2 // 2)] -= W12[0:(n2 // 2), 0:(l2 // 2)]
+            self += work
+            work[0:n2, 0:l2] = 0
+
+            #M2
+            CheapMatrix.multiply(W21,(A21 + A22),(B11), W22)
+            W22[0:(n2 // 2), 0:(l2 // 2)] -= W21[0:(n2 // 2), 0:(l2 // 2)]
+            self += work
+            work[0:n2, 0:l2] = 0
+
+            #M1
+            CheapMatrix.multiply(W11,(A11 + A22), (B11 + B22), W22)
+            W22[0:(n2 // 2), 0:(l2 // 2)] = W11[0:(n2 // 2), 0:(l2 // 2)]
+            W21[0:W21.nrow(), 0:W21.ncol()] = 0
+            W12[0:W12.nrow(), 0:W12.ncol()] = 0
+            self += work
+            work[0:n2, 0:l2] = 0
+
+            #odvečni deli, ki niso deli bloka
+            #print(n, m,l)
+            for n1 in range(n):
+                for l1 in range(l):
+                    for m1 in range(m):
+                        #print(n1,n-n2,"nji",m1, m-m2,"mji", l1, l - l2, "lji")
+                        if not (n1 < n2 and m1 < m2 and l1 < l2):
+                            self[n1, l1] += left[n1, m1] * right[m1, l1]
+            return self
+
+
+
+
```

Comment:
Testov ne vključuj v module, saj se bodo tako pognali ob uvozu. Če hočeš imeti svoje teste, naredi mapo `JureMarkun` v mapi `test` in jih premakni tja.